aws/resource_aws_api_gateway_vpc_link.go

```diff
@@ -0,0 +1,200 @@
+package aws
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsApiGatewayVpcLink() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsApiGatewayVpcLinkCreate,
+		Read:   resourceAwsApiGatewayVpcLinkRead,
+		Update: resourceAwsApiGatewayVpcLinkUpdate,
+		Delete: resourceAwsApiGatewayVpcLinkDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"target_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayVpcLinkCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	input := &apigateway.CreateVpcLinkInput{
+		Name:       aws.String(d.Get("name").(string)),
+		TargetArns: []*string{aws.String(d.Get("target_arn").(string))},
+	}
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+
+	resp, err := conn.CreateVpcLink(input)
+	if err != nil {
+		return err
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{apigateway.VpcLinkStatusPending},
+		Target:     []string{apigateway.VpcLinkStatusAvailable},
+		Refresh:    apigatewayVpcLinkRefreshStatusFunc(conn, *resp.Id),
+		Timeout:    5 * time.Minute,
```

Comment:
I was trying this PR out, and hit a case where this operation timed out after 5 minutes.  The VPC Link did end up being successfully created.  The AWS docs note that this can take "2-4 minutes", so it may be worth allowing more buffer here (and on update), to ensure these don't time out under normal conditions?

Reply:
Thank you for your advice!